src/datasources/asset-calibration/AssetCalibrationDataSource.ts

```diff
@@ -34,9 +36,15 @@ export class AssetCalibrationDataSource extends DataSourceBase<AssetCalibrationQ
 
   baseUrl = this.instanceSettings.url + '/niapm/v1';
 
+  state: AssetCalibrationDataSourceState = {
+    systems: null
+  }
```

Comment:
I'm not sure I see the value of the 'state' type here. I'd rather have something like:
`systemAliasCache: Map<string, SystemMetadata>`

Reply:
Also we may want to consider setting a TTL on the cache. Not sure how often we expect aliases to change, but we do this in the data frame plugin.
https://github.com/ni/systemlink-grafana-plugins/blob/dc1725296a6211fd49974cd56fac257b51b4a4b2/src/datasources/data-frame/DataFrameDataSource.ts#L23

Reply:
I've read the docs on https://www.npmjs.com/package/@isaacs/ttlcache. It looks like it's great for individual items.  Given that we work with a collection I think this code will work fine.